app/upgrades/v15/upgrades.go

```diff
@@ -18,6 +20,113 @@ import (
+
+// Using the same queries as stargate_whitelist.go
+var whitelistedQueries = []string{
+	// cosmos-sdk queries
+
+	// auth
+	"/cosmos.auth.v1beta1.Query/Account",
+	"/cosmos.auth.v1beta1.Query/Params",
+
+	// bank
+	"/cosmos.bank.v1beta1.Query/Balance",
+	"/cosmos.bank.v1beta1.Query/DenomMetadata",
+	"/cosmos.bank.v1beta1.Query/Params",
+	"/cosmos.bank.v1beta1.Query/SupplyOf",
+
+	// distribution
+	"/cosmos.distribution.v1beta1.Query/Params",
+	"/cosmos.distribution.v1beta1.Query/DelegatorWithdrawAddress",
+	"/cosmos.distribution.v1beta1.Query/ValidatorCommission",
+
+	// gov
+	"/cosmos.gov.v1beta1.Query/Deposit",
+	"/cosmos.gov.v1beta1.Query/Params",
+	"/cosmos.gov.v1beta1.Query/Vote",
+
+	//slashing
+	"/cosmos.slashing.v1beta1.Query/Params",
+	"/cosmos.slashing.v1beta1.Query/SigningInfo",
+
+	//staking
+	"/cosmos.staking.v1beta1.Query/Delegation",
+	"/cosmos.staking.v1beta1.Query/Params",
+	"/cosmos.staking.v1beta1.Query/Validator",
+
+	//osmosis queries
+
+	//epochs
+	"/osmosis.epochs.v1beta1.Query/EpochInfos",
+	"/osmosis.epochs.v1beta1.Query/CurrentEpoch",
+
+	//gamm
+	"/osmosis.gamm.v1beta1.Query/NumPools",
+	"/osmosis.gamm.v1beta1.Query/TotalLiquidity",
+	"/osmosis.gamm.v1beta1.Query/Pool",
+	"/osmosis.gamm.v1beta1.Query/PoolParams",
+	"/osmosis.gamm.v1beta1.Query/TotalPoolLiquidity",
+	"/osmosis.gamm.v1beta1.Query/TotalShares",
+	"/osmosis.gamm.v1beta1.Query/CalcJoinPoolShares",
+	"/osmosis.gamm.v1beta1.Query/CalcExitPoolCoinsFromShares",
+	"/osmosis.gamm.v1beta1.Query/CalcJoinPoolNoSwapShares",
+	"/osmosis.gamm.v1beta1.Query/PoolType",
+	"/osmosis.gamm.v2.Query/SpotPrice",
+	"/osmosis.gamm.v1beta1.Query/EstimateSwapExactAmountIn",
+	"/osmosis.gamm.v1beta1.Query/EstimateSwapExactAmountOut",
+
+	//incentives
+	"/osmosis.incentives.Query/ModuleToDistributeCoins",
+	"/osmosis.incentives.Query/LockableDurations",
+
+	//lockup
+	"/osmosis.lockup.Query/ModuleBalance",
+	"/osmosis.lockup.Query/ModuleLockedAmount",
+	"/osmosis.lockup.Query/AccountUnlockableCoins",
+	"/osmosis.lockup.Query/AccountUnlockingCoins",
+	"/osmosis.lockup.Query/LockedDenom",
+	"/osmosis.lockup.Query/LockedByID",
+
+	//mint
+	"/osmosis.mint.v1beta1.Query/EpochProvisions",
+	"/osmosis.mint.v1beta1.Query/Params",
+
+	//pool-incentives
+	"/osmosis.poolincentives.v1beta1.Query/GaugeIds",
+
+	//superfluid
+	"/osmosis.superfluid.Query/Params",
+	"/osmosis.superfluid.Query/AssetType",
+	"/osmosis.superfluid.Query/AllAssets",
+	"/osmosis.superfluid.Query/AssetMultiplier",
+
+	//poolmanager
+	"/osmosis.poolmanager.v1beta1.Query/NumPools",
+	"/osmosis.poolmanager.v1beta1.Query/EstimateSwapExactAmountIn",
+	"/osmosis.poolmanager.v1beta1.Query/EstimateSwapExactAmountOut",
+
+	//txfees
+	"/osmosis.txfees.v1beta1.Query/FeeTokens",
+	"/osmosis.txfees.v1beta1.Query/DenomSpotPrice",
+	"/osmosis.txfees.v1beta1.Query/DenomPoolId",
+	"/osmosis.txfees.v1beta1.Query/BaseDenom",
+
+	//tokenfactory
+	"/osmosis.tokenfactory.v1beta1.Query/params",
```

Comment:
I know this is from stargate whitelist but is it actually correct to be lowercase?

For example, here it's upper case: https://github.com/osmosis-labs/osmosis/blob/9f1e7ef8920afac2015f3e0aef4094fd31b8de4b/x/tokenfactory/types/query.pb.go#L386

Reply:
I would guess it isn't. I'll test it and change it in both places if it idn't.

Reply:
I think it does matter as there is a pure Go string comparison:
https://github.com/strangelove-ventures/async-icq/blob/6f0b046511d24936943d44c890b9e392dcd32d50/types/keys.go#L29

As discussed offline, going to make a PR to this branch with the change